### PR TITLE
Lock down the operator process security context

### PR DESCRIPTION
**Pull request by sdake**
1. Ensure the filesystem is readonly
2. Disable any privilege escalation
3. Drop all capabilities

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Review by morvencao — COMMENTED** [review]
Not sure if we also need to update https://github.com/istio/istio/blob/master/operator/deploy/operator.yaml
otherwise lgtm.

File: manifests/charts/istio-operator/templates/deployment.yaml [hunk 1]
```diff
@@ -20,6 +20,13 @@ spec:
           command:
           - operator
           - server
+          securityContext:
```
Comment (howardjohn) [inline manifests/charts/istio-operator/templates/deployment.yaml:23]: Discovery specifies:
```
          securityContext:
            runAsUser: 1337
            runAsGroup: 1337
            runAsNonRoot: true
            capabilities:
              drop:
              - ALL
```

My understanding here is a "correct" way to do this, based on some obscure details of how PSP's mutate pods, that I am not 100% aware of. Maybe @dgn or @rlenglet can shed some light. I would expect  these to probably look the same, but the correct answer may be a mix of these two

Comment (rlenglet) [reply manifests/charts/istio-operator/templates/deployment.yaml:23]: Agreed. It's better to specify all the fields in `securityContext` to lock down the pod and avoid surprises with PSPs. So I would specify:
```
    securityContext:
      allowPrivilegeEscalation: false
      capabilities:
        drop:
        - ALL
      privileged: false
      readOnlyRootFilesystem: true
      runAsGroup: 1337
      runAsUser: 1337
      runAsNonRoot: true
```

**Review by rlenglet — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

**Review by morvencao — APPROVED** [review]

**Pull request merged**